Assets/00_Content/Scripts/Interactables/PickUp.cs

```diff
@@ -1,14 +1,38 @@
-﻿using UnityEngine;
+﻿using Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
 
 namespace Interactables {
 
 	public class PickUp : MonoBehaviour {
 
 		[SerializeField] private Transform itemGrabTransform;
+		[SerializeField] private Collider objectCollider;
+		[SerializeField] private LayerMask itemSlotLayer;
 
+		//Drop item on floor or snap to slot if close
 		public void DropItem() {
+
 			transform.SetParent(null);
+			Collider[] collisions = Physics.OverlapBox(transform.position, transform.localScale / 2, Quaternion.identity);
```

Comment:
What if my pickup is not a box? And what if I use a different sized box mesh without changing the scale?